utils/big_decimal.cc

```diff
@@ -143,6 +143,35 @@ std::strong_ordering big_decimal::operator<=>(const big_decimal& other) const
     return x.compare(y) <=> 0;
 }
 
+std::strong_ordering big_decimal::operator<=>(const big_decimal& other) const
+{
+    if (_scale == other._scale) {
```

Comment:
Even before: compare the sign function on the two mantissas, if they are different, or if both are zero, that's the result.

Reply:
Yes, I forgot about the sign.

Reply:
Done in v2.